js/app.js

```diff
@@ -48,10 +50,18 @@ $(document).ready(function() {
 
   // Input binding
   $searchInput
-  .on('keyup', function() {
-    var query = $(this).val();
-    toggleIconEmptyInput(query);
-    algoliaHelper.setQuery(query).search();
+  .on('keyup paste cut change', function() {
+    var $el = $(this);
+
+    // We use setTimeout to defer the callback to the bottom of the stack to
+    // catch the value after the paste, cut, or delete via mouse
+    setTimeout(function(){
+      var query = $el.val();
```

Comment:
@vvo The reason for the `setTimeout` is that in the location in the stack where the event is triggered, the value hasn't been updated. I will be honest that I don't know the reason why exactly.

*However*, the `input propertychange` works perfectly--no need to leave the input box to register the event. Great suggestion.